Minor fix

httpsnet opened the pull request:
Fixed a display issue with #settingstab.
I updated the HTML of each resource only when the textContent changed. This should significantly reduce the page refresh process.
Fixed an issue with achievement messages.
Added "date" to promotion code.
The promotion code allows uppercase letters. The button text in Special Actions should work as is.
spoiler added octeracts per second.

KhafraDev commented on src/ImportExport.ts:
```diff
@@ -475,7 +478,14 @@ export const promocodes = async (input: string | null) => {
-        return Alert(`You will gain an octeract (when they come out) every ${format(1 / perSecond, 2, true)} seconds, assuming you have them unlocked!`)
+        const perSecondText = perSecond >= 1 ? `${format(perSecond, 2, true)} per second` : `every ${format(1 / perSecond, 2, true)} seconds`
+        return Alert(`You will gain an octeract (when they come out) ${perSecondText}, assuming you have them unlocked!`)
+    } else if (input === 'date') {
```
is this needed?

httpsnet replied:
Before Cx20 was nerfed, I've been able to get less than 0.1 with normal data.
Therefore, I think it may progress to 0 in future versions.
Ascension Speed is too powerful and I think it should be reduced by exponentiation.
If it's still not needed, I'll erase the code I pointed out.

httpsnet replied:
I got 0.72 seconds with spoiler at 10/14/14/14/14/14/14/14 of main data sing 33 in v2.9.8.
Upload data if necessary.
Should the date code be erased? I think Synergism needs a watch.
The following ImportExport.ts upload contains more fixes.

KhafraDev replied:
I don't think adding this as a promocode is the solution

httpsnet replied:
ok, deleted

httpsnet commented on src/ImportExport.ts:
```diff
@@ -421,22 +424,25 @@ export const promocodes = async (input: string | null) => {
         const availableUses = timeCodeAvailableUses();
         if (availableUses === 0) {
-            return Confirm(`
+            return Alert(`
```
All the questions you point out are confusing to users, as the results are the same for yes and no.
When I actually confirmed it by the user's report, I understood it.

httpsnet replied:
I didn't get a reply, so I accepted your request and resolved it

httpsnet commented on src/ImportExport.ts:
```diff
@@ -424,22 +424,25 @@ export const promocodes = async (input: string | null) => {
         if (availableUses === 0) {
-            return Confirm(`
+            return Alert(`
             If you imported a save, you cannot use this code for 15 minutes to prevent cheaters.
```
Here, fraudster measures aren't really working.
Should I take measures like gamble's localStorage?

KhafraDev commented on src/ImportExport.ts:
```diff
@@ -427,16 +432,18 @@ export const promocodes = async (input: string | null) => {
+            'Click the OK button within the next 15 seconds to test your luck!' +
             ` If you click within ${format(2500 + 125 * player.cubeUpgrades[61], 0, true)} ms of a randomly generated time, you will win a prize!` +
-            ` This particular instance has a ${format(rewardMult, 2, true)}x multiplier due to elapsed time between uses.`
+            ` This particular instance has a x${format(rewardMult, 2, true)} multiplier due to elapsed time between uses.`
```
x should be at the back
"5.5x"
"x5.5"

httpsnet replied:
ok

httpsnet commented:
I fixed more. sorry.

KhafraDev commented on src/Toggles.ts:
```diff
@@ -47,6 +47,9 @@ export const toggleTabs = (name: keyof typeof tabNumberConst) => {
     revealStuff();
     hideStuff();
 
+    const el = document.activeElement as HTMLElement;
```
this is nullable

httpsnet replied:
ok
"eslint-disable-next-line @typescript-eslint/no-unnecessary-condition" is also gone.

KhafraDev commented on src/Toggles.ts:
```diff
@@ -321,16 +324,22 @@ export const keyboardTabChange = (dir = 1, main = true) => {
+    const subTabs = subTabsInMainTab(mainTab)
+    if (tabs(mainTab).unlocked && subTabs.subTabList.length > 0) {
+
+        const el = document.activeElement as HTMLElement;
```
ditto

httpsnet commented:
I forgot to write the update contents of ebb13ed

Fixed an issue where all cube upgrades would be initialized when reloading after singularity and before ascension
This was the cause of the loss of w1x7 earned in updateSingularityMilestoneAwards().
Added w1x7 with singularity 20. Added w1x10 in singularity 11.
Auto Ascend fixed an issue that could actually be done just by enabling the toggle.
